src/System-SessionManager/SessionAccessModeResolver.class.st

```diff
@@ -0,0 +1,49 @@
+"
+I check the image arguments and resolve if the access mode of the image. I am started very soon in the image start-up process.
+"
+Class {
+	#name : #SessionAccessModeResolver,
+	#superclass : #Object,
+	#category : #'System-SessionManager-Utilities'
+}
+
+{ #category : #'class initialization' }
+SessionAccessModeResolver class >> initialize [
+
+	"self initialize"
+
+	SessionManager default
+		register: (ClassSessionHandler forClassNamed: self name)
+		inCategory: SessionManager default systemCategory  
+		atPriority: SessionManager default defaultPriority - 90.
+]
+
+{ #category : #'class initialization' }
+SessionAccessModeResolver class >> resolve [ 
+
+	"check if the image was started in the read-only mode"
+
+	| arguments mode | 
+	
+	arguments := Smalltalk argumentsStartingAtIndex: 0.
+			
+	mode := #readWrite.
+
+	(arguments anySatisfy: [ :each | each = '--readWriteAccessMode' ])
+		ifTrue: [ mode := #readWrite ].
+	(arguments anySatisfy: [ :each | each = '--readOnlyAccessMode' ])
+		ifTrue: [ mode := #readOnly ].
+	(arguments anySatisfy: [ :each | each = '--writeOnlyAccessMode' ])
+		ifTrue: [ mode := #writeOnly ].
+	(arguments anySatisfy: [ :each | each = '--disabledAccessMode' ])
+		ifTrue: [ mode := #disabled ].
```

Comment:
I don't want to be annoying :). But is not it too premature implementation of those modes? Here we need only one.
If if will have many of them it could be better to reify real objects instead of symbols. So we can put behaviour to them

Reply:
we have two so currently, it is only two more ;-) When I was writing the resolver, I just wanted to make it complete. I had the first-class access mode in mind but it looked not important to me